libpromises/generic_agent.c

```diff
@@ -446,6 +446,20 @@ void LoadAugments(EvalContext *ctx, GenericAgentConfig *config)
     free(def_json);
 }
 
+void AddPolicyEntryVariables (EvalContext *ctx, GenericAgentConfig *config)
+{
+    char *abs_input_path = GetAbsolutePath(config->input_file);
+    /* both dirname() and basename() may actually modify the string they are given (see man:basename(3)) */
+    char *dirname_path = xstrdup(abs_input_path);
+    char *basename_path = xstrdup(abs_input_path);
+    EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "policy_entry_filename", abs_input_path, CF_DATA_TYPE_STRING, "source=agent");
+    EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "policy_entry_dirname", dirname(dirname_path), CF_DATA_TYPE_STRING, "source=agent");
+    EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "policy_entry_basename", basename(basename_path), CF_DATA_TYPE_STRING, "source=agent");
+    free (abs_input_path);
```

Comment:
remove the space before parenthesis, and try to wrap at 78. See HACKING.md.

Reply:
Done.